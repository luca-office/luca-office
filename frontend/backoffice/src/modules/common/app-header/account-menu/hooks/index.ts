export * from "./use-account-menu"
