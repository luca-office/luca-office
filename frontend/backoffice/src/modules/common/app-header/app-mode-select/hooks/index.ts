export * from "./use-app-mode-select"
