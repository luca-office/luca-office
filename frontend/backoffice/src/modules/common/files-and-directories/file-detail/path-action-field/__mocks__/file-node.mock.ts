import {FileNode} from "shared/components"
import {BinaryType, NodeType} from "shared/enums"

export const fileNodeMock: FileNode = {
  binaryFileId: "spfosdkfsd",
  binaryFileUrl: "url@url.de",
  binaryType: BinaryType.Image,
  id: "sfpoksdf-fposjfd",
  name: "test-file",
  parentId: "sfopksdf-fsdpfosdk",
  tags: [],
  type: NodeType.File
}
