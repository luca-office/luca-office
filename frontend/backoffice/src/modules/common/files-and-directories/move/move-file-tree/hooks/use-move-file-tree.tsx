import {useState} from "react"

export interface UseMoveFileTreeHook {
  readonly expandedDirectories: string[]
  readonly toggleExpandedDirectory: (id: UUID) => void
}

export const useMoveFileTree = (): UseMoveFileTreeHook => {
  const [expandedDirectories, setExpandedDirectories] = useState<string[]>([])

  const toggleExpandedDirectory = (id: UUID) =>
    setExpandedDirectories(
      expandedDirectories.includes(id) ? expandedDirectories.filter(id => id !== id) : expandedDirectories.concat(id)
    )

  return {
    expandedDirectories,
    toggleExpandedDirectory
  }
}
