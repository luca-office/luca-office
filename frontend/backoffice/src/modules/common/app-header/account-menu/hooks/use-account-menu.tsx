import i18next from "i18next"
import {useCallback, useState} from "react"
import {useDispatch} from "react-redux"
import {useCheckLogin} from "shared/graphql/hooks"
import {LucaAppLanguage, UserAccountForLogin} from "shared/models"
import {Option} from "shared/utils"
import {useLogout} from "../../../../../graphql/hooks"
import {navigateToRouteAction} from "../../../../../redux/actions/navigation-action"
import {Route} from "../../../../../routes"

export interface UseAccountMenuHook {
  readonly isMenuVisible: boolean
  readonly activeLanguage: string
  readonly logout: () => void
  readonly openAccountSettings: () => void
  readonly toggleMenu: () => void
  readonly userAccount: Option<UserAccountForLogin>
  readonly setStoredSelectedLanguage: (language: LucaAppLanguage) => void
}

export const useAccountMenu = (): UseAccountMenuHook => {
  const [isMenuVisible, toggleMenuVisibility] = useState(false)
  const {logout} = useLogout()
  const {account: userAccount} = useCheckLogin()
  const dispatch = useDispatch()

  const openAccountSettings = useCallback(() => {
    dispatch(navigateToRouteAction(Route.MyAccount))
  }, [])

  const toggleMenu = () => {
    toggleMenuVisibility(!isMenuVisible)
  }

  const handleSetStoredSelectedLanguage = (language: LucaAppLanguage) => i18next.changeLanguage(language)

  const activeLanguage = i18next.language

  return {
    isMenuVisible,
    activeLanguage,
    logout,
    openAccountSettings,
    toggleMenu,
    userAccount,
    setStoredSelectedLanguage: handleSetStoredSelectedLanguage
  }
}
