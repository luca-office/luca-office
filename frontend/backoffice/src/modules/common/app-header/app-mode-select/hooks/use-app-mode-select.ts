import {useCallback, useState} from "react"
import {useDispatch, useSelector} from "react-redux"
import {useCheckLogin} from "shared/graphql/hooks"
import {UserAccountForLogin} from "shared/models"
import {Option} from "shared/utils"
import {AppMode} from "../../../../../enums"
import {navigateToRouteAction} from "../../../../../redux/actions/navigation-action"
import {updateAppMode} from "../../../../../redux/actions/ui/common-ui-action"
import {AppState} from "../../../../../redux/state/app-state"
import {Route} from "../../../../../routes"

export interface UseAppModeSelectHook {
  readonly appMode: AppMode
  readonly changeAppMode: (mode: AppMode) => void
  readonly isMenuVisible: boolean
  readonly toggleMenu: () => void
  readonly account: Option<UserAccountForLogin>
}

export const useAppModeSelect = (): UseAppModeSelectHook => {
  const [isMenuVisible, toggleMenuVisibility] = useState(false)
  const appMode = useSelector<AppState, AppMode>(state => state.ui.common.appMode)

  const {account} = useCheckLogin()
  const dispatch = useDispatch()

  const changeAppMode = useCallback((appMode: AppMode) => {
    dispatch(updateAppMode(appMode))

    switch (appMode) {
      case AppMode.MANAGER:
        dispatch(navigateToRouteAction(Route.Projects))
        break
      case AppMode.USERMANAGEMENT:
        dispatch(navigateToRouteAction(Route.UserManagement))
        break
      case AppMode.RATING:
        dispatch(navigateToRouteAction(Route.RaterRatingOverview))
        break
      case AppMode.EDITOR:
      default:
        dispatch(navigateToRouteAction(Route.Scenarios))
    }
    toggleMenuVisibility(false)
  }, [])

  const toggleMenu = () => {
    toggleMenuVisibility(!isMenuVisible)
  }

  return {
    appMode,
    changeAppMode,
    isMenuVisible,
    toggleMenu,
    account
  }
}
