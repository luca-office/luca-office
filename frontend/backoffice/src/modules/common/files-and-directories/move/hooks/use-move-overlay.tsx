import {Dispatch, SetStateAction, useState} from "react"
import {FileType} from "shared/enums"
import {useUpdateFile} from "shared/graphql/hooks"
import {Directory, File} from "shared/models"
import {findPathToRoot, Option} from "shared/utils"
import {useUpdateDirectory} from "../../../../../graphql/hooks/mutations/directory/update-directory"

export interface UseMoveOverlay {
  readonly selectedTargetDirectory: Option<UUID>
  readonly setSelectedTargetDirectory: Dispatch<SetStateAction<Option<string>>>
  readonly changeDirectory: () => void
}

// Use this if you need an id for root and can not use null
export const ROOT = "root_identifier"

export const useMoveOverlay = (
  directoryOption: Option<Directory>,
  fileOption: Option<File>,
  onFilesSuccessfullyMoved: (targetDirectoryIdWithParents: UUID[]) => void,
  directories: Directory[],
  files: File[]
): UseMoveOverlay => {
  const [selectedTargetDirectory, setSelectedTargetDirectory] = useState<Option<UUID>>(Option.none())

  const {updateDirectory} = useUpdateDirectory()
  const {updateFile} = useUpdateFile()

  const handleChangeDirectory = () => {
    directoryOption.forEach(directory =>
      selectedTargetDirectory.forEach(targetDirectory => {
        const parentDirectoryId = targetDirectory === ROOT ? null : targetDirectory

        const parentDirectoryIdWithParents = getTargetDirectoryIdsWithParents(targetDirectory, directories, files)
        updateDirectory(directory.id, {name: directory.name, parentDirectoryId}).then(() =>
          onFilesSuccessfullyMoved(parentDirectoryIdWithParents)
        )
      })
    )

    fileOption.forEach(file =>
      selectedTargetDirectory.forEach(targetDirectory => {
        const parentDirectoryIdWithParents = getTargetDirectoryIdsWithParents(targetDirectory, directories, files)
        updateFile(file.id, {
          binaryFileId: file.fileType === FileType.Media ? file.binaryFileId : null,
          name: file.name,
          relevance: file.relevance,
          tags: file.tags,
          directoryId: targetDirectory
        }).then(() => onFilesSuccessfullyMoved(parentDirectoryIdWithParents))
      })
    )
  }

  return {
    selectedTargetDirectory,
    setSelectedTargetDirectory,
    changeDirectory: handleChangeDirectory
  }
}

const getTargetDirectoryIdsWithParents = (targetDirectoryId: UUID, directories: Directory[], files: File[]) => {
  const pathToRoot = findPathToRoot(targetDirectoryId, directories, files)
  return [...pathToRoot, targetDirectoryId]
}
