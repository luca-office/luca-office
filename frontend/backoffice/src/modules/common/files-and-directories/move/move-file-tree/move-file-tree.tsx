import {css} from "@emotion/react"
import * as React from "react"
import {FileExplorer, Heading, Icon, RenderDirectoryProps, Text, Tooltip} from "shared/components"
import {HeadingLevel, IconName} from "shared/enums"
import {Directory, DirectoryNode, File} from "shared/models"
import {
  backgroundColorLight,
  borderRadius,
  fileExplorerSelectedColor,
  Flex,
  FontWeight,
  primaryColor,
  spacing,
  spacingCard,
  spacingLarge,
  spacingMedium,
  spacingSmall,
  spacingTiny,
  TextSize
} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {createTree, findPathToNode, Option} from "shared/utils"
import {CustomDirectoryNode} from "../custom-directory-node/custom-directory-node"
import {ROOT} from "../hooks/use-move-overlay"
import {useMoveFileTree} from "./hooks/use-move-file-tree"

export interface MoveFileTreeProps {
  readonly isFile: boolean
  readonly currentParentDirectory: Option<DirectoryNode>
  readonly currentDirectoryId: Option<UUID>
  readonly selectedTargetDirectory: Option<UUID>
  readonly setSelectedTargetDirectory: React.Dispatch<React.SetStateAction<Option<string>>>
  readonly dataLoading: boolean
  readonly directories: Directory[]
  readonly files: File[]
}

export const MoveFileTree: React.FC<MoveFileTreeProps> = ({
  isFile,
  currentParentDirectory,
  selectedTargetDirectory,
  setSelectedTargetDirectory,
  currentDirectoryId,
  dataLoading: isLoading,
  directories,
  files
}) => {
  const {expandedDirectories, toggleExpandedDirectory} = useMoveFileTree()

  const tree = createTree(directories, files)

  const {t} = useLucaTranslation()

  const renderDirectoryNode = (options: RenderDirectoryProps) => (
    <CustomDirectoryNode
      currentDirectoryId={currentDirectoryId}
      currentParentDirectory={currentParentDirectory}
      isChildOfOrigin={isChildOfOrigin(
        currentDirectoryId,
        options.node.id,
        directories,
        currentParentDirectory.map(directory => directory.id)
      )}
      selectedTargetDirectory={selectedTargetDirectory}
      setSelectedTargetDirectory={setSelectedTargetDirectory}
      node={options.node}
      onClickChevron={options.onClickChevron}
      isDirectoryExpanded={options.isDirectoryExpanded}
      isDirectorySelected={options.isDirectorySelected}
      level={options.level}
    />
  )

  const renderCustomMoveToRootEntry = () => {
    const isSelected = selectedTargetDirectory.exists(target => target === ROOT)

    return (
      <div
        className="custom-move-to-root"
        css={moveFileTreeStyles.moveToRootEntry.container(isSelected)}
        onClick={isSelected ? undefined : () => setSelectedTargetDirectory(Option.of(ROOT))}>
        <Icon customStyles={moveFileTreeStyles.moveToRootEntry.icon} name={IconName.HardDrive} />
        <Text size={TextSize.Small}>{t("files_and_directories__move_modal_define_as_root_directory")}</Text>
        {isSelected && (
          <Tooltip title={t("files_and_directories__move_modal_tooltip_revert_selection")}>
            <Icon
              className="custom-move-to-root-close-button"
              customStyles={moveFileTreeStyles.moveToRootEntry.closeButton(isSelected)}
              name={IconName.Close}
              onClick={() => setSelectedTargetDirectory(Option.none())}
            />
          </Tooltip>
        )}
      </div>
    )
  }

  return (
    <div css={[Flex.column, {flex: 1}]}>
      <Heading customStyles={moveFileTreeStyles.title} level={HeadingLevel.h3} fontWeight={FontWeight.Bold}>
        {t("files_and_directories__move_modal_select_directory")}
      </Heading>
      <FileExplorer
        customStyles={{
          card: [moveFileTreeStyles.card, Flex.column],
          cardContent: {boxShadow: "inset 0px 0px 4px 0px rgba(0, 0, 0, 0.16)"}
        }}
        tree={tree}
        expandedDirectoryIds={expandedDirectories}
        isLoading={isLoading}
        onExpandDirectory={toggleExpandedDirectory}
        selectedNodeId={Option.none()}
        renderOnlyDirectories
        renderDirectoryNode={renderDirectoryNode}
        isCreateDirectoryButtonVisible={false}
        renderCustomTopEntry={!isFile && currentParentDirectory.isDefined() ? renderCustomMoveToRootEntry : undefined}
      />
    </div>
  )
}

const moveFileTreeStyles = {
  card: css({
    width: "100%",
    boxSizing: "border-box",
    flex: "1 1 0",
    overflow: "auto"
  }),
  content: css({
    boxShadow: "inset 0px 0px 4px 0px rgba(0, 0, 0, 0.16)"
  }),
  title: css({
    marginTop: spacingMedium,
    marginBottom: spacingTiny
  }),
  moveToRootEntry: {
    container: (isSelected: boolean) =>
      css(Flex.row, {
        boxSizing: "border-box",
        boxShadow: isSelected ? `${spacing(0, 1, 2, 0)} rgba(0, 0, 0, 0.24)` : "none",
        cursor: isSelected ? "default" : "pointer",
        padding: spacing(spacingTiny, spacingCard, spacingTiny, spacingMedium + spacingTiny),
        borderRadius: borderRadius,
        height: spacingLarge,
        border: `1px solid ${isSelected ? primaryColor : backgroundColorLight}`,
        marginBottom: spacingCard,

        ":hover": {
          backgroundColor: fileExplorerSelectedColor
        }
      }),
    closeButton: (isSelected: boolean) =>
      css({
        cursor: isSelected ? "pointer" : "inherit",
        marginLeft: "auto"
      }),
    icon: css({marginRight: spacingSmall})
  }
}

const isChildOfOrigin = (
  currentDirectoryId: Option<string>,
  selectedNode: string,
  directories: Directory[],
  currentParentDirectoryId: Option<string>
) =>
  currentDirectoryId.exists(directoryId =>
    findPathToNode(selectedNode, directories, [], currentParentDirectoryId.orUndefined()).includes(directoryId)
  )
