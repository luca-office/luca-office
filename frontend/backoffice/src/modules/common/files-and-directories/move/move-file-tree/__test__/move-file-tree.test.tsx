// importing from direct file because of issues of babel loader and spyOn
import {MockedProvider} from "@apollo/client/testing"
import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {FileExplorer, Heading} from "shared/components"
import {NodeType} from "shared/enums"
import {filesMock} from "shared/graphql/__mocks__"
import {DirectoryNode} from "shared/models"
import {Option} from "shared/utils"
import {directoriesForScenarioMock} from "../../../../../../graphql/__mocks__"
import {ROOT} from "../../hooks/use-move-overlay"
import * as useMoveFileTree from "../hooks/use-move-file-tree"
import {UseMoveFileTreeHook} from "../hooks/use-move-file-tree"
import {MoveFileTree, MoveFileTreeProps} from "../move-file-tree"

const hookValuesDefault: UseMoveFileTreeHook = {
  expandedDirectories: [],
  toggleExpandedDirectory: jest.fn()
}

const stateSpy = jest.spyOn(useMoveFileTree, "useMoveFileTree")

const defaultProps: MoveFileTreeProps = {
  isFile: false,
  currentParentDirectory: Option.none(),
  currentDirectoryId: Option.none(),
  selectedTargetDirectory: Option.none(),
  setSelectedTargetDirectory: jest.fn(),
  dataLoading: false,
  directories: directoriesForScenarioMock,
  files: filesMock
}

const mockParentDirectory: DirectoryNode = {
  id: "6b183a2e-0af6-4006-95b5-e2daca778616",
  parentId: "b6897871-96be-4927-894d-ee99b4a8f30a",
  name: "Urlaub",
  type: NodeType.Directory,
  children: []
}

const useComponentWithProvider = (props?: Partial<MoveFileTreeProps>) => (
  <MockedProvider>
    <MoveFileTree {...defaultProps} {...props} />
  </MockedProvider>
)

const useComponent = (props?: Partial<MoveFileTreeProps>) => <MoveFileTree {...defaultProps} {...props} />

describe("Move File Tree", () => {
  it("renders correctly", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)
    const component = create(useComponentWithProvider())

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure - all options empty", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)

    const component = shallow(useComponent(), {wrappingComponent: MockedProvider})
    expect(component.find(Heading)).toHaveLength(1)
    expect(component.find(FileExplorer)).toHaveLength(1)
  })

  it("renders root target option if selected directory has a parent", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)

    const component = shallow(
      useComponent({
        currentParentDirectory: Option.of(mockParentDirectory)
      }),
      {
        wrappingComponent: MockedProvider
      }
    )

    expect(component.find(FileExplorer).dive().find(".custom-move-to-root")).toHaveLength(1)
    expect(component.find(FileExplorer).dive().find(".custom-move-to-root-close-button")).toHaveLength(0)
  })

  it("renders correctly if root is selected as target", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)

    const component = shallow(
      useComponent({
        selectedTargetDirectory: Option.of(ROOT),
        currentParentDirectory: Option.of(mockParentDirectory)
      }),
      {
        wrappingComponent: MockedProvider
      }
    )

    expect(component.find(FileExplorer).dive().find(".custom-move-to-root")).toHaveLength(1)
    expect(component.find(FileExplorer).dive().find(".custom-move-to-root-close-button")).toHaveLength(1)
  })
})
