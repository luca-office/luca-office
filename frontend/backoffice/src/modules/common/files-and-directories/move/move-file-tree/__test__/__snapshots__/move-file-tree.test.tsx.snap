// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Move File Tree renders correctly 1`] = `
<div
  className="css-22pday-MoveFileTree"
>
  <div
    className="css-12ezatt-Heading"
  >
    files_and_directories__move_modal_select_directory
  </div>
  <div
    className="css-tzta12-Card"
  >
    <div
      className="card-header css-15zaw2t-CardHeader"
    >
      <div
        className="css-mv0otu-TableOfContentsContainer"
      >
        <div
          className="css-oivmdv-Icon"
          dangerouslySetInnerHTML={
            {
              "__html": {},
            }
          }
        />
        <div
          className="css-1f6p2vj-Heading"
        >
          file_explorer__title
        </div>
      </div>
    </div>
    <div
      className="toc-body css-iiw7p1-TableOfContentsContainer"
    >
      <div
        className="book-chapters-list css-1n23nny-TableOfContentsContainer"
      >
        <div
          className="css-18xeog2-TableOfContentsContainer"
        >
          <div
            className="css-1j6baqz-CustomDirectoryNode"
            data-testid="custom-directory-move"
            onClick={[Function]}
          >
            <div
              className="css-1x436ap-CustomDirectoryNode"
            >
              <div
                className="css-1ep6va6-CustomDirectoryNode"
              >
                <div
                  className="css-0"
                  id="tooltip_42"
                >
                  <div
                    className="css-1a0pk7x-CustomDirectoryNode"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                    onClick={[Function]}
                  />
                </div>
                <div
                  className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                  role="tooltip"
                  style={
                    {
                      "backgroundColor": "#2e3032",
                      "zIndex": 1,
                    }
                  }
                >
                  
                  <div
                    className="react-tooltip-arrow styles-module_arrow__K0L3T"
                    style={{}}
                  />
                </div>
                <div
                  className="css-1sn69or-CustomDirectoryNode"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1axqtld-Text"
                >
                  Bilder (3)
                </div>
              </div>
              <div
                className="move-text css-16u4ufw-CustomDirectoryNode"
              >
                <div
                  className="css-1uyodls-Text"
                >
                  files_and_directories__move_modal_target_directory
                </div>
                <div
                  className="css-g4lidz-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="css-1j6baqz-CustomDirectoryNode"
            data-testid="custom-directory-move"
            onClick={[Function]}
          >
            <div
              className="css-1x436ap-CustomDirectoryNode"
            >
              <div
                className="css-1ep6va6-CustomDirectoryNode"
              >
                <div
                  className="css-0"
                  id="tooltip_42"
                >
                  <div
                    className="css-1a0pk7x-CustomDirectoryNode"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                    onClick={[Function]}
                  />
                </div>
                <div
                  className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                  role="tooltip"
                  style={
                    {
                      "backgroundColor": "#2e3032",
                      "zIndex": 1,
                    }
                  }
                >
                  
                  <div
                    className="react-tooltip-arrow styles-module_arrow__K0L3T"
                    style={{}}
                  />
                </div>
                <div
                  className="css-1sn69or-CustomDirectoryNode"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1axqtld-Text"
                >
                  Rechnungen (3)
                </div>
              </div>
              <div
                className="move-text css-16u4ufw-CustomDirectoryNode"
              >
                <div
                  className="css-1uyodls-Text"
                >
                  files_and_directories__move_modal_target_directory
                </div>
                <div
                  className="css-g4lidz-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            className="css-1j6baqz-CustomDirectoryNode"
            data-testid="custom-directory-move"
            onClick={[Function]}
          >
            <div
              className="css-1x436ap-CustomDirectoryNode"
            >
              <div
                className="css-1ep6va6-CustomDirectoryNode"
              >
                <div
                  className="css-0"
                  id="tooltip_42"
                >
                  <div
                    className="css-dbigyp-CustomDirectoryNode"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                    onClick={[Function]}
                  />
                </div>
                <div
                  className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                  role="tooltip"
                  style={
                    {
                      "backgroundColor": "#2e3032",
                      "zIndex": 1,
                    }
                  }
                >
                  files_and_directories__move_modal_no_subfolders
                  <div
                    className="react-tooltip-arrow styles-module_arrow__K0L3T"
                    style={{}}
                  />
                </div>
                <div
                  className="css-1sn69or-CustomDirectoryNode"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                <div
                  className="css-1axqtld-Text"
                >
                  Sonstiges (0)
                </div>
              </div>
              <div
                className="move-text css-16u4ufw-CustomDirectoryNode"
              >
                <div
                  className="css-1uyodls-Text"
                >
                  files_and_directories__move_modal_target_directory
                </div>
                <div
                  className="css-g4lidz-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card-footer css-105jaxv-CardFooter"
    />
  </div>
</div>
`;
